src/main/java/org/gridsuite/sensitivityanalysis/server/dto/nonevacuatedenergy/NonEvacuatedEnergyInputData.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.sensitivityanalysis.server.dto.nonevacuatedenergy;
+
+import com.powsybl.sensitivity.SensitivityAnalysisParameters;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.SuperBuilder;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@SuperBuilder
+@NoArgsConstructor
+@Getter
+@Setter
+@Schema(description = "Sensitivity analysis non evacuated energy input data")
+public class NonEvacuatedEnergyInputData {
+    private List<NonEvacuatedEnergyStageDefinition> nonEvacuatedEnergyStagesDefinition;
+
+    private List<NonEvacuatedEnergyStagesSelection> nonEvacuatedEnergyStagesSelection;
```

Comment:
It's related to the comment in the tests but I would change the data model so it does not rely on lists and index of lists. It makes relationships between objects clearer and avoid to iterate through indexes in the code.